examples/htool.c

```diff
@@ -251,6 +251,413 @@ static int do_target_reset(uint32_t reset_option) {
+    int offset = sizeof(struct payload_status_response_header) +
+        region_index * sizeof(struct payload_region_state);
+    const struct payload_region_state* rs =
+        (struct payload_region_state*)(&buffer[offset]);
+    printf("Region %c:\n", region_index == 0 ? 'A' : 'B');
+    printf("  validation_state: %s (%u)\n",
+           PayloadValidationStateString(rs->validation_state),
+           rs->validation_state);
+    printf("  failure_reason: %s (%u)\n",
+           PayloadValidationFailureReasonString(rs->failure_reason),
+           rs->failure_reason);
+    printf("  image_type: %s (%u)\n", ImageTypeString(rs->image_type),
+           rs->image_type);
+    printf("  key_index:  %u\n", rs->key_index);
+    printf("  image_family: (0x%08x)\n", rs->image_family);
+    printf("  version: %u.%u.%u.%u\n", rs->version_major, rs->version_minor,
+           rs->version_point, rs->version_subpoint);
+    printf("  descriptor_offset: 0x%08x\n", rs->descriptor_offset);
+  }
+  return 0;
+}
+
+static int command_payload_version(const struct htool_invocation* inv) {
+  
+}
+
+size_t size_to_skip(const uint8_t* data, size_t max_size) {
+  for (int i = 0; i < max_size; i++) {
+    if (data[i] != 0xff) return i;
+  }
+  return max_size;
+}
+
+size_t size_to_send(const uint8_t* data, size_t size) {
+  for (int i = size; i > 0; i--) {
+    if (data[i - 1] != 0xff) return i;
+  }
+  return 0;
+}
+
+struct payload_update_packet {
+  uint32_t offset; /* image offset */
+  uint32_t len;    /* packet length excluding this header */
+  uint8_t type;    /* One of PAYLOAD_UPDATE_* */
+  /* payload data immediately follows */
+} __attribute__((packed));
+
+static int send_payload_update_command(struct libhoth_usb_device* dev,
+                                       uint8_t command, uint32_t offset,
+                                       uint32_t len) {
+  struct payload_update_packet request;
+  request.offset = offset;
+  request.len = len;
+  request.type = command;
+
+  uint8_t response[1];  // ignored
+  size_t rlen = 0;
+
+  int ret = htool_exec_hostcmd(
+      dev, EC_CMD_BOARD_SPECIFIC_BASE + EC_PRV_CMD_HOTH_PAYLOAD_UPDATE, 0,
+      &request, sizeof(request), &response, sizeof(response), &rlen);
+  return ret;
+}
+
+static int send_payload_update_command_with_return_value(struct libhoth_usb_device* dev,
+                                       uint8_t command, uint32_t offset,
+                                       uint32_t len, uint8_t* buf) {
+  struct payload_update_packet request;
+  request.offset = offset;
+  request.len = len;
+  request.type = command;
+  size_t rlen = 0;
+
+  int ret = htool_exec_hostcmd(
+      dev, EC_CMD_BOARD_SPECIFIC_BASE + EC_PRV_CMD_HOTH_PAYLOAD_UPDATE, 0,
+      &request, sizeof(request), buf, len, &rlen);
+  return ret;
+}
+
+int command_payload_read(struct libhoth_usb_device* dev, uint32_t offset, uint32_t len, uint8_t* buf) {
+  return send_payload_update_command_with_return_value(dev, 6, offset, len, buf);
```

Comment:
`6` is a magic number here -- the subcommands opcodes should be in a enum instead, especially if we're using them in multiple places.

(Same issue elsewhere, but I haven't noted every such case, to avoid comment spam.)

Reply:
Sure, will remove payload-related code for now, and address them in a separate change.